include/mgvalue.hpp

```diff
@@ -0,0 +1,1604 @@
+#pragma once
+
+#include <cstring>
+#include <initializer_list>
+#include <set>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+#include <glog/logging.h>
+
+#include "mgclient.h"
+
+namespace mg {
+
+namespace detail {
+/// \brief Type castign using memcpy.
+/// uint to int conversion in C++ is a bit tricky. Take a look here
+/// https://stackoverflow.com/questions/14623266/why-cant-i-reinterpret-cast-uint-to-int
+/// for more details.
+///
+/// \tparam TDest Returned datatype.
+/// \tparam TSrc Input datatype.
+///
+/// \return "copy casted" value.
+///
```

Comment:
This isn't part of the public API, you can remove the formal Doxygen documentation and just leave the note as an ordinary comment (`//`).